advanced_alchemy/repository/_async.py

```diff
@@ -169,6 +193,9 @@ async def add(
         """
         with wrap_sqlalchemy_exception():
             instance = await self._attach_to_session(data)
+            if self._load:
+                await self._flush_or_commit(auto_commit=True)
```

Comment:
Should the `auto_commit` follow what was sent in from the method or actually be `True`?

Reply:
I think it should stay `True` since the following `self._refresh_with_load()` emits a select to get back the newly inserted rows with loaded relationships.

Reply:
Just thinking through this a bit more, and it's still not totally clear.

We definitely need a flush so that the inserted row is loaded into relationship.  As long as it isn't a new session though, we don't have to commit to make that happen.  However, i'm not quite following why the commit is necessary.

Is there a simple use case that I can walk through to visualize why the commit over a flush is needed?